### PR TITLE
Fixes to "No primary found..." issue for mongo

### LINKED ISSUE #1658: No primary found in replicaset or invalid replica set name.
My existing MongoDB database shows this error 

image

Creating a new database shows that error as well

image

1. Create a Cosmos DB account with the MongoDB API
2. Once created the database is unusable with the above error

The below info is what happens when you attempt to create a database within the account.

---

Action: cosmosDB.createMongoDatabase
Error type: MongoError
Error Message: no primary found in replicaset or invalid replica set name

Version: 0.14.0
OS: win32
OS Release: 10.0.19041
Product: Visual Studio Code - Insiders
Product Version: 1.48.0-insider
Language: en

<details>
<summary>Call Stack</summary>

```
replset.js:624:11mongodb/replset.js:624:11
connectNewServers mongodb/replset.js:279:12
topologyMonitor mongodb/replset.js:611:12
Server.<anonymous> mongodb/replset.js:841:7
Object.onceWrapper events.js:291:20
Server.emit events.js:203:13
Pool.<anonymous> mongodb/server.js:384:12
Pool.emit events.js:203:13
pool.js:580:12mongodb/pool.js:580:12
pool.js:1035:7mongodb/pool.js:1035:7
callback mongodb/connect.js:97:5
connect.js:397:5mongodb/connect.js:397:5
auth_provider.js:61:11mongodb/auth_provider.js:61:11
_callback mongodb/connect.js:349:5
Connection.messageHandler mongodb/connect.js:378:5
Connection.emit events.js:203:13
processMessage mongodb/connection.js:384:10
TLSSocket.<anonymous> mongodb/connection.js:553:15
TLSSocket.emit events.js:203:13
addChunk _stream_readable.js:295:12
readableAddChunk _stream_readable.js:276:11
TLSSocket.Readable.push _stream_readable.js:210:10
TLSWrap.onStreamRead stream_base_commons.js:166:17
```

</details>

### PR DESCRIPTION
**Pull request by neelip**
Fixes #1658

### CODE REVIEW

File: src/mongo/connectToMongoClient.ts [hunk 1]
```diff
@@ -12,7 +12,9 @@ export async function connectToMongoClient(connectionString: string, appName: st
         // appName should be wrapped in '@'s when trying to connect to a Mongo account, this doesn't effect the appendUserAgent string
         appName: `@${appName}@`,
         // https://github.com/lmammino/mongo-uri-builder/issues/2
-        useNewUrlParser: true
+        useNewUrlParser: true,
+        useUnifiedTopology: true,
+        serverSelectionTimeoutMS: 1000
```
Comment (EricJizbaMSFT) [inline src/mongo/connectToMongoClient.ts:17]: I'm gonna need more explanation for your changes

Comment (neelip) [reply src/mongo/connectToMongoClient.ts:17]: The property `useUnifiedTopology` will be set to true by default from the next version mongodb. 
This property being true resolves the issue of "No primary found in replicaset or invalid replica set name." which appeared in the Nightly Build earlier (now resolved with recent updates) and is related to issue #1658 

It also resolves this warning... 
`DeprecationWarning: current Server Discovery and Monitoring engine is
deprecated, and will be removed in a future version. To use the new Server
Discover and Monitoring engine, pass option { useUnifiedTopology: true } to
the MongoClient constructor.
`

https://mongodb.github.io/node-mongodb-native/3.3/reference/unified-topology/
image

image

image

Comment (nturinski) [reply src/mongo/connectToMongoClient.ts:17]: Why not use the default `serverSelectionTimeoutMS`?

Comment (neelip) [reply src/mongo/connectToMongoClient.ts:17]: @nturinski actually that would work too, I didn't notice the default time earlier. @ejizba What do you think?

Comment (ejizba) [reply src/mongo/connectToMongoClient.ts:17]: Yeah I would stick with the default unless you have some specific reason you want to change it

**Review by ejizba — APPROVED** [review]

**Pull request merged**